common/src/crypto.rs

```diff
@@ -32,20 +44,45 @@ impl fmt::Display for Hash256 {
 /// A cryptographic signature.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
 pub struct Signature {
-    pub dummy: String,
+    signature: Vec<u8>,
 }
 
 impl Signature {
     /// Creates a new signature from the given data and keys.
-    pub fn sign(_data: Hash256, _public_key: &PublicKey, _private_key: &PrivateKey) -> Self {
-        Signature {
-            dummy: "".to_string(),
-        }
+    pub fn sign(
+        data: Hash256,
+        public_key: &PublicKey,
+        private_key: &PrivateKey,
+    ) -> Result<Self, Error> {
+        let public_key = ed25519_dalek::PublicKey::from_bytes(&public_key.key)
+            .map_err(|_| Error::InvalidFormat(format!("public key: {}", public_key)))?;
+        let private_key = ed25519_dalek::SecretKey::from_bytes(&private_key.key)
+            .map_err(|_| Error::InvalidFormat("private key: [omitted]".to_owned()))?;
```

Comment:
I think we need to check keypair match before we sign the message.
Otherwise, it will result in an invalid signature.

Reply:
fixed.